include/envoy/ratelimit/ratelimit.h

```diff
@@ -34,5 +38,49 @@ struct Descriptor {
   absl::optional<RateLimitOverride> limit_ = absl::nullopt;
 };
 
+/**
+ * A single token bucket. See token_bucket.proto.
+ */
+struct TokenBucket {
+  uint32_t max_tokens_;
+  uint32_t tokens_per_fill_;
+  std::chrono::milliseconds fill_interval_;
+
+  friend bool operator==(const TokenBucket& lhs, const TokenBucket& rhs) {
+    return lhs.max_tokens_ == rhs.max_tokens_ && lhs.tokens_per_fill_ == rhs.tokens_per_fill_ &&
+           lhs.fill_interval_ == rhs.fill_interval_;
+  }
+
+  // Support absl::Hash.
+  template <typename H>
+  friend H AbslHashValue(H h, const TokenBucket& d) { // NOLINT(readability-identifier-naming)
+    h = H::combine(std::move(h), d.max_tokens_, d.tokens_per_fill_);
+    return h;
+  }
+};
+
+/**
+ * A single rate limit request descriptor. See ratelimit.proto.
+ */
+struct LocalDescriptor {
+  std::vector<DescriptorEntry> entries_;
+  TokenBucket token_bucket_ = {};
+
+  friend bool operator==(const LocalDescriptor& lhs, const LocalDescriptor& rhs) {
+    return lhs.entries_ == rhs.entries_ && lhs.token_bucket_ == rhs.token_bucket_;
+  }
+
+  // Support absl::Hash.
+  template <typename H>
+  friend H AbslHashValue(H h, const LocalDescriptor& d) { // NOLINT(readability-identifier-naming)
```

Comment:
You should be hashing on token bucket also if you use it in equality above.

Reply:
Verified that I use config descriptors to the match for request allowed and not the populated route descriptors. Hence using token bucket in hash too now.